nifi-nar-bundles/nifi-mongodb-bundle/nifi-mongodb-processors/src/test/java/org/apache/nifi/processors/mongodb/GetMongoRecordIT.java

```diff
@@ -0,0 +1,208 @@
+        int index = 0;
+        for (Map<String, Object> it : parsed) {
+            assertEquals(values.get(index++), it.get("failedLogins"));
+            assertNull(it.get("name"));
+            assertNull(it.get("lastLogin"));
+        }
+    }
+
+    private static interface LookupValidator {
+        void validate(TestRunner runner);
+    }
+
+    @Test
+    void testLookup() throws Exception {
+        LookupValidator ffValidator =  (runner) -> {
+            List<MockFlowFile> ffs = runner.getFlowFilesForRelationship(GetMongoRecord.REL_SUCCESS);
+            assertNotNull(ffs);
+            assertTrue(ffs.size() == 1);
+            assertEquals("3", ffs.get(0).getAttribute("record.count"));
+            assertEquals("application/json", ffs.get(0).getAttribute(CoreAttributes.MIME_TYPE.key()));
+            assertEquals(COL_NAME, ffs.get(0).getAttribute(GetMongoRecord.COL_NAME));
+            assertEquals(DB_NAME, ffs.get(0).getAttribute(GetMongoRecord.DB_NAME));
+            assertEquals(Document.parse("{}"), Document.parse(ffs.get(0).getAttribute("executed.query")));
+        };
+
+        Map<String, String> attrs = new HashMap<>();
+        attrs.put("schema.name", "sample");
+
+        runner.setProperty(GetMongoRecord.QUERY_ATTRIBUTE, "executed.query");
+        runner.setProperty(GetMongoRecord.QUERY, "{}");
+        runner.enqueue("", attrs);
+        runner.run();
+
+        runner.assertTransferCount(GetMongoRecord.REL_FAILURE, 0);
+        runner.assertTransferCount(GetMongoRecord.REL_SUCCESS, 1);
+        runner.assertTransferCount(GetMongoRecord.REL_ORIGINAL, 1);
+
+        ffValidator.validate(runner);
+
+        runner.clearTransferState();
+        runner.removeProperty(GetMongoRecord.QUERY);
+        runner.enqueue("{}", attrs);
+        runner.run();
+
+        runner.assertTransferCount(GetMongoRecord.REL_FAILURE, 0);
+        runner.assertTransferCount(GetMongoRecord.REL_SUCCESS, 1);
+        runner.assertTransferCount(GetMongoRecord.REL_ORIGINAL, 1);
+
+        ffValidator.validate(runner);
+    }
+
+    @Test
+    void testLimit() throws Exception {
+        runner.setIncomingConnection(false);
+        runner.setProperty(GetMongoRecord.LIMIT, "1");
+        runner.setProperty(GetMongoRecord.QUERY, "{}");
+        runner.setVariable("schema.name", "sample");
+        runner.run();
+
+        List<Map<String, Object>> parsed = sharedTest();
+        assertEquals(1, parsed.size());
+    }
+
+    List<Map<String, Object>> sharedTest() throws Exception {
+        runner.assertTransferCount(GetMongoRecord.REL_FAILURE, 0);
+        runner.assertTransferCount(GetMongoRecord.REL_SUCCESS, 1);
+
+        MockFlowFile ff = runner.getFlowFilesForRelationship(GetMongoRecord.REL_SUCCESS).get(0);
+        byte[] raw = runner.getContentAsByteArray(ff);
+        String content = new String(raw);
+        List<Map<String, Object>> parsed = new ObjectMapper().readValue(content, List.class);
```

Comment:
`ObjectMapper` should be declared once as a member variable and reused.

Reply:
Done.